keras_nlp/models/task.py

```diff
@@ -267,59 +274,162 @@ def from_preset(
+            cls = subclasses[0]
+        # Forward dtype to the backbone.
+        config_overrides = {}
+        if "dtype" in kwargs:
+            config_overrides["dtype"] = kwargs.pop("dtype")
+        backbone = load_from_preset(
+            preset,
+            load_weights=load_weights,
+            config_overrides=config_overrides,
+        )
+        if "preprocessor" in kwargs:
+            preprocessor = kwargs.pop("preprocessor")
+        else:
+            tokenizer = load_from_preset(
                 preset,
-                load_weights=load_weights,
-                config_overrides=config_overrides,
+                config_file=TOKENIZER_CONFIG_FILE,
             )
-            if "preprocessor" in kwargs:
-                preprocessor = kwargs.pop("preprocessor")
-            else:
-                tokenizer = load_from_preset(
-                    preset,
-                    config_file="tokenizer.json",
-                )
-                preprocessor = cls.preprocessor_cls(tokenizer=tokenizer)
+            preprocessor = cls.preprocessor_cls(tokenizer=tokenizer)
+
+        # TODO: Do we expect remote paths here? os.path doesn't work for remote paths. replacements: https://github.com/keras-team/keras/blob/master/keras/utils/file_utils.py
+        task_config_path = os.path.join(preset, TASK_CONFIG_FILE)
+        if not os.path.exists(task_config_path):
             return cls(backbone=backbone, preprocessor=preprocessor, **kwargs)
 
-        # Task case.
-        if not issubclass(preset_cls, cls):
+        # TODO: I should probably move task loading logic to preset_utils.py?
+        # Load Task from preset.
+        if not issubclass(cls, Task):
+            raise ValueError(
+                "`{cls.__name__}` should be subclass of Task!"
+            )  # TODO: update error message
+        task_config_class = check_config_class(
+            preset, config_file=TASK_CONFIG_FILE
+        )
+        if not issubclass(task_config_class, cls):
+            raise ValueError(
+                f"`{TASK_CONFIG_FILE}` has type `{task_config_class.__name__}` "
+                f"which is not a subclass of calling class `{cls.__name__}`. Call "
+                f"`from_preset` directly on `{task_config_class.__name__}` instead."
+            )
+
+        task_config_file = os.path.join(preset, TASK_CONFIG_FILE)
+        with open(task_config_file, "r") as config_file:
+            task_config = json.load(config_file)
+        # TODO: add back backbone and preprocessor config when save_to_preset removes them (rn, save_to_preset, doesn't remove them!).
+        # task_config.update(backbone_config)
+        # task_config.update(preprocessor_config)
+        task = keras.saving.deserialize_keras_object(task_config)
+        load_weights = load_weights and task_config["weights"]
+        task_weights_path = os.path.join(preset, task_config["weights"])
+        task.load_task_weights(task_weights_path)
```

Comment:
We need to call `task.backbone.load_weights` too somewhere right? Where do the backbone weights get loaded?

Reply:
Backbone is loaded before task loading by calling `load_from_preset` [here](https://github.com/keras-team/keras-nlp/pull/1547/files#diff-32e324081e133371b2d59a60e6622050b67f3517674b83e788604a2efad43ef2R311-R315).

Then if preset has `task.json`, task is loaded and we set `task.backbone = backbone` [here](https://github.com/keras-team/keras-nlp/pull/1547/files#diff-32e324081e133371b2d59a60e6622050b67f3517674b83e788604a2efad43ef2R360).